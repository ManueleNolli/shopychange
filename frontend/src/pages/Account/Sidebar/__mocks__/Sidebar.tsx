import { divWithoutChildrenMock } from '../../../../../mocksUtils'

const SidebarMock = () => divWithoutChildrenMock('Sidebar')
export default SidebarMock
