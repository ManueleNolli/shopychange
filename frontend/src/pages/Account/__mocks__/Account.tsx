import { divWithoutChildrenMock } from '../../../../mocksUtils'

const AccountMock = () => divWithoutChildrenMock('Account')
export default AccountMock
