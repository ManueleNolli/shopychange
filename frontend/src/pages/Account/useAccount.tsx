import { useMediaQuery } from '@chakra-ui/react'
import { useNavigate } from 'react-router-dom'
import { SidebarData } from '../../data/accountSidebarData'

export default function useAccount() {
  const navigate = useNavigate()
  // Media query for mobile
  const [isMobile] = useMediaQuery('(max-width: 768px)')

  const changeActivePage = (element: SidebarData) => {
    navigate(`/account/dashboard/${element.route}`)
  }

  return {
    isMobile,
    changeActivePage,
  }
}
