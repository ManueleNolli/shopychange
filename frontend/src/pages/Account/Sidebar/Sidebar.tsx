import React, { useEffect } from 'react'

import {
  Box,
  Text,
  Flex,
  Divider,
  Center,
  useColorModeValue,
  useDisclosure,
  Button,
  VStack,
  HStack,
} from '@chakra-ui/react'
import { motion } from 'framer-motion'

type SidebarProps = {
  data: SidebarDataWithActive[]
  children: React.ReactNode
  direction?: 'row' | 'column'
  onClick: (element: SidebarData) => void
}
import { SidebarData } from '../../../data/accountSidebarData'

import { FaAngleLeft, FaAngleRight } from 'react-icons/fa'

import { colors } from '../../../styles/Colors'
import { SidebarDataWithActive } from '../Account'

export default function Sidebar({
  data,
  children,
  direction = 'row',
  onClick,
}: SidebarProps) {
  const { getButtonProps, getDisclosureProps, isOpen, onToggle } =
    useDisclosure()

  useEffect(() => {
    if (!isOpen) onToggle()
  }, [])

  return (
    <Flex direction={direction}>
      {direction === 'row' ? (
        <motion.div
          {...getDisclosureProps()}
          initial={false}
          hidden={false}
          animate={{ width: isOpen ? 400 : 100 }}
          style={{
            whiteSpace: 'nowrap',
            height: '100%',
          }}
        >
          <Box
            m="2"
            bg={useColorModeValue('white', 'gray.800')}
            boxShadow={'xl'}
            rounded={'lg'}
          >
            <Flex
              justifyContent={isOpen ? 'flex-end' : 'center'}
              alignItems={'center'}
              pt="4"
            >
              <Button
                {...getButtonProps()}
                variant={'unstyled'}
                data-testid="toggle-sidebar"
              >
                <Center>
                  {isOpen ? (
                    <FaAngleLeft size="24px" />
                  ) : (
                    <FaAngleRight size="24px" />
                  )}
                </Center>
              </Button>
            </Flex>

            <VStack
              pb={isOpen ? '10' : '5'}
              mx={isOpen ? '5' : '2'}
              alignItems={isOpen ? 'start' : 'center'}
              spacing="12px"
              divider={isOpen ? <Divider /> : <></>}
            >
              {data.map((item) => {
                return (
                  <RenderElementRow
                    isActive={item.active}
                    data={data}
                    key={item.text}
                    {...item}
                    onClick={onClick}
                    textVisible={isOpen}
                  />
                )
              })}
            </VStack>
          </Box>
        </motion.div>
      ) : (
        <Center my="5">
          <HStack spacing="12px" divider={<Divider />}>
            {data.map((item) => {
              return (
                <Center
                  data-testid="item-row-onclick"
                  onClick={() => onClick(item)}
                  p="4"
                  rounded={'md'}
                  role="group"
                  bg={item.active ? '#EBF8FFAA' : 'transparent'}
                  _hover={{
                    cursor: 'pointer',
                    bg: useColorModeValue(
                      `${colors.popover.bgChakraColor}.50`,
                      'gray.900'
                    ),
                    color: `${colors.popover.bgChakraColor}.400`,
                  }}
                  transition={'color .3s ease, transform .3s ease'}
                  key={item.text}
                >
                  <Center
                    _groupHover={{ transform: 'scale(1.33)' }}
                    transition={'transform .3s ease'}
                  >
                    {item.icon}
                  </Center>
                </Center>
              )
            })}
          </HStack>
        </Center>
      )}

      <Box w="100%">{children}</Box>
    </Flex>
  )
}

type SidebarElementRowProps = {
  isActive: boolean
  data: SidebarData[]
  icon: React.ReactNode
  text: string
  textVisible: boolean
  onClick: (element: SidebarData) => void
}
function RenderElementRow({
  isActive,
  data,
  icon,
  text,
  textVisible,
  onClick,
}: SidebarElementRowProps) {
  return (
    <Flex
      ml={textVisible ? '10%' : '0'}
      py="5"
      px="2"
      rounded={'md'}
      onClick={() => {
        data.forEach((item: SidebarData) => {
          if (item.text === text) onClick(item)
        })
      }}
      role={'group'}
      minW={textVisible ? '80%' : '0'}
      bg={isActive ? '#EBF8FFAA' : 'transparent'}
      _hover={{
        cursor: 'pointer',
        bg: useColorModeValue(`${colors.popover.bgChakraColor}.50`, 'gray.900'),
        color: `${colors.popover.bgChakraColor}.400`,
      }}
      transition={'color .3s ease, transform .3s ease'}
    >
      <Center
        flex={textVisible ? '0' : '1'}
        _groupHover={{ transform: 'scale(1.33)' }}
        transition={'transform .3s ease'}
      >
        {icon}
      </Center>
      <Text ml="12px" hidden={!textVisible}>
        {text}
      </Text>
    </Flex>
  )
}
