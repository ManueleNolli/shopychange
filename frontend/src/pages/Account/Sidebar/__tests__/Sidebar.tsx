import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Sidebar from '../Sidebar'
import { TbPhoto, TbPhotoStar, TbPhotoCog } from 'react-icons/tb'
import { Text } from '@chakra-ui/react'
import { SidebarDataWithActive } from '../../Account'

const mockedData: SidebarDataWithActive[] = [
  {
    icon: <TbPhoto size="24px" />,
    text: 'My NFTs',
    route: 'my-nfts',
    component: <Text> My NFTs </Text>,
    active: true,
  },
  {
    icon: <TbPhotoStar size="24px" />,
    text: 'Favorites NFTs',
    route: 'favourite-nfts',
    component: <Text> Favourite </Text>,
    active: false,
  },
  {
    icon: <TbPhotoCog size="24px" />,
    text: 'Observed Collections',
    route: 'observed-collections',
    component: <Text> Observed Collections </Text>,
    active: false,
  },
]
const mockOnClick = jest.fn()

describe('Sidebar', () => {
  test('should render the Sidebar component row open', () => {
    const tree = render(
      <Sidebar data={mockedData} onClick={mockOnClick}>
        <div data-testid="children">children</div>
      </Sidebar>
    )
    expect(tree.getByText('My NFTs')).toBeInTheDocument()
    expect(tree.getByText('Favorites NFTs')).toBeInTheDocument()
    expect(tree.getByText('Observed Collections')).toBeInTheDocument()
    expect(tree.getByTestId('children')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('should render the Sidebar component row close', () => {
    const tree = render(
      <Sidebar data={mockedData} onClick={mockOnClick}>
        <div data-testid="children">children</div>
      </Sidebar>
    )

    fireEvent.click(tree.getByTestId('toggle-sidebar'))

    expect(tree.container).toMatchSnapshot()
  })

  test('should render the Sidebar component columns', () => {
    const tree = render(
      <Sidebar data={mockedData} onClick={mockOnClick} direction="column">
        <div data-testid="children">children</div>
      </Sidebar>
    )
    expect(tree.getByTestId('children')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('should call onClick row', () => {
    const tree = render(
      <Sidebar data={mockedData} onClick={mockOnClick}>
        <div data-testid="children">children</div>
      </Sidebar>
    )
    fireEvent.click(tree.getByText('Favorites NFTs'))
    expect(mockOnClick).toHaveBeenCalled()
    expect(tree.container).toMatchSnapshot()
  })
  test('should call onClick column', () => {
    const tree = render(
      <Sidebar data={mockedData} onClick={mockOnClick} direction="column">
        <div data-testid="children">children</div>
      </Sidebar>
    )
    fireEvent.click(tree.getAllByTestId('item-row-onclick')[0])
    expect(mockOnClick).toHaveBeenCalled()
    expect(tree.container).toMatchSnapshot()
  })
})
