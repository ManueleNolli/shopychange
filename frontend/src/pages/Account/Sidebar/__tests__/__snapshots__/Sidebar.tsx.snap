// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should call onClick column 1`] = `
<div>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-8977ob"
    >
      <div
        class="chakra-stack css-1p1m4ay"
      >
        <div
          class="css-1o4fl00"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M3 6a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v12a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3v-12z"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l5 5"
              />
              <path
                d="M14 14l1 -1c.928 -.893 2.072 -.893 3 0l3 3"
              />
            </svg>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-lg8s77"
        />
        <div
          class="css-1ssbfw8"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M11 21h-5a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v5.5"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l2 2"
              />
              <path
                d="M17.8 20.817l-2.172 1.138a.392 .392 0 0 1 -.568 -.41l.415 -2.411l-1.757 -1.707a.389 .389 0 0 1 .217 -.665l2.428 -.352l1.086 -2.193a.392 .392 0 0 1 .702 0l1.086 2.193l2.428 .352a.39 .39 0 0 1 .217 .665l-1.757 1.707l.414 2.41a.39 .39 0 0 1 -.567 .411l-2.172 -1.138z"
              />
            </svg>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-lg8s77"
        />
        <div
          class="css-1ssbfw8"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M12 21h-6a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v6"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l3 3"
              />
              <path
                d="M14 14l1 -1c.48 -.461 1.016 -.684 1.551 -.67"
              />
              <path
                d="M19.001 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M19.001 15.5v1.5"
              />
              <path
                d="M19.001 21v1.5"
              />
              <path
                d="M22.032 17.25l-1.299 .75"
              />
              <path
                d="M17.27 20l-1.3 .75"
              />
              <path
                d="M15.97 17.25l1.3 .75"
              />
              <path
                d="M20.733 20l1.3 .75"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8atqhb"
    >
      <div
        data-testid="children"
      >
        children
      </div>
    </div>
  </div>
</div>
`;

exports[`Sidebar should call onClick row 1`] = `
<div>
  <div
    class="css-1xhj18k"
  >
    <div
      id="disclosure-:r3:"
      style="white-space: nowrap; height: 100%; width: 100px;"
    >
      <div
        class="css-16inpur"
      >
        <div
          class="css-krjmb"
        >
          <button
            aria-controls="disclosure-:r3:"
            aria-expanded="true"
            class="chakra-button css-4xx2wk"
            data-testid="toggle-sidebar"
            type="button"
          >
            <div
              class="css-gmuwbf"
            >
              <svg
                fill="currentColor"
                height="24px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="chakra-stack css-1aad1bn"
        >
          <div
            class="css-8b158v"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M3 6a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v12a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3v-12z"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l5 5"
                />
                <path
                  d="M14 14l1 -1c.928 -.893 2.072 -.893 3 0l3 3"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              My NFTs
            </p>
          </div>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1ej9zno"
          />
          <div
            class="css-21y338"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M11 21h-5a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v5.5"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l2 2"
                />
                <path
                  d="M17.8 20.817l-2.172 1.138a.392 .392 0 0 1 -.568 -.41l.415 -2.411l-1.757 -1.707a.389 .389 0 0 1 .217 -.665l2.428 -.352l1.086 -2.193a.392 .392 0 0 1 .702 0l1.086 2.193l2.428 .352a.39 .39 0 0 1 .217 .665l-1.757 1.707l.414 2.41a.39 .39 0 0 1 -.567 .411l-2.172 -1.138z"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              Favorites NFTs
            </p>
          </div>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1ej9zno"
          />
          <div
            class="css-21y338"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M12 21h-6a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v6"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l3 3"
                />
                <path
                  d="M14 14l1 -1c.48 -.461 1.016 -.684 1.551 -.67"
                />
                <path
                  d="M19.001 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
                />
                <path
                  d="M19.001 15.5v1.5"
                />
                <path
                  d="M19.001 21v1.5"
                />
                <path
                  d="M22.032 17.25l-1.299 .75"
                />
                <path
                  d="M17.27 20l-1.3 .75"
                />
                <path
                  d="M15.97 17.25l1.3 .75"
                />
                <path
                  d="M20.733 20l1.3 .75"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              Observed Collections
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8atqhb"
    >
      <div
        data-testid="children"
      >
        children
      </div>
    </div>
  </div>
</div>
`;

exports[`Sidebar should render the Sidebar component columns 1`] = `
<div>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-8977ob"
    >
      <div
        class="chakra-stack css-1p1m4ay"
      >
        <div
          class="css-1o4fl00"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M3 6a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v12a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3v-12z"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l5 5"
              />
              <path
                d="M14 14l1 -1c.928 -.893 2.072 -.893 3 0l3 3"
              />
            </svg>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-lg8s77"
        />
        <div
          class="css-1ssbfw8"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M11 21h-5a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v5.5"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l2 2"
              />
              <path
                d="M17.8 20.817l-2.172 1.138a.392 .392 0 0 1 -.568 -.41l.415 -2.411l-1.757 -1.707a.389 .389 0 0 1 .217 -.665l2.428 -.352l1.086 -2.193a.392 .392 0 0 1 .702 0l1.086 2.193l2.428 .352a.39 .39 0 0 1 .217 .665l-1.757 1.707l.414 2.41a.39 .39 0 0 1 -.567 .411l-2.172 -1.138z"
              />
            </svg>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-lg8s77"
        />
        <div
          class="css-1ssbfw8"
          data-testid="item-row-onclick"
          role="group"
        >
          <div
            class="css-17mrdb"
          >
            <svg
              fill="none"
              height="24px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M15 8h.01"
              />
              <path
                d="M12 21h-6a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v6"
              />
              <path
                d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l3 3"
              />
              <path
                d="M14 14l1 -1c.48 -.461 1.016 -.684 1.551 -.67"
              />
              <path
                d="M19.001 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M19.001 15.5v1.5"
              />
              <path
                d="M19.001 21v1.5"
              />
              <path
                d="M22.032 17.25l-1.299 .75"
              />
              <path
                d="M17.27 20l-1.3 .75"
              />
              <path
                d="M15.97 17.25l1.3 .75"
              />
              <path
                d="M20.733 20l1.3 .75"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8atqhb"
    >
      <div
        data-testid="children"
      >
        children
      </div>
    </div>
  </div>
</div>
`;

exports[`Sidebar should render the Sidebar component row close 1`] = `
<div>
  <div
    class="css-1xhj18k"
  >
    <div
      id="disclosure-:r1:"
      style="white-space: nowrap; height: 100%; width: 100px;"
    >
      <div
        class="css-16inpur"
      >
        <div
          class="css-13v5be6"
        >
          <button
            aria-controls="disclosure-:r1:"
            aria-expanded="false"
            class="chakra-button css-4xx2wk"
            data-testid="toggle-sidebar"
            type="button"
          >
            <div
              class="css-gmuwbf"
            >
              <svg
                fill="currentColor"
                height="24px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="chakra-stack css-1dmcozd"
        >
          <div
            class="css-pt1d8w"
            role="group"
          >
            <div
              class="css-2432ay"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M3 6a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v12a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3v-12z"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l5 5"
                />
                <path
                  d="M14 14l1 -1c.928 -.893 2.072 -.893 3 0l3 3"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
              hidden=""
            >
              My NFTs
            </p>
          </div>
          <div
            class="css-hzhtyx"
            role="group"
          >
            <div
              class="css-2432ay"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M11 21h-5a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v5.5"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l2 2"
                />
                <path
                  d="M17.8 20.817l-2.172 1.138a.392 .392 0 0 1 -.568 -.41l.415 -2.411l-1.757 -1.707a.389 .389 0 0 1 .217 -.665l2.428 -.352l1.086 -2.193a.392 .392 0 0 1 .702 0l1.086 2.193l2.428 .352a.39 .39 0 0 1 .217 .665l-1.757 1.707l.414 2.41a.39 .39 0 0 1 -.567 .411l-2.172 -1.138z"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
              hidden=""
            >
              Favorites NFTs
            </p>
          </div>
          <div
            class="css-hzhtyx"
            role="group"
          >
            <div
              class="css-2432ay"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M12 21h-6a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v6"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l3 3"
                />
                <path
                  d="M14 14l1 -1c.48 -.461 1.016 -.684 1.551 -.67"
                />
                <path
                  d="M19.001 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
                />
                <path
                  d="M19.001 15.5v1.5"
                />
                <path
                  d="M19.001 21v1.5"
                />
                <path
                  d="M22.032 17.25l-1.299 .75"
                />
                <path
                  d="M17.27 20l-1.3 .75"
                />
                <path
                  d="M15.97 17.25l1.3 .75"
                />
                <path
                  d="M20.733 20l1.3 .75"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
              hidden=""
            >
              Observed Collections
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8atqhb"
    >
      <div
        data-testid="children"
      >
        children
      </div>
    </div>
  </div>
</div>
`;

exports[`Sidebar should render the Sidebar component row open 1`] = `
<div>
  <div
    class="css-1xhj18k"
  >
    <div
      id="disclosure-:r0:"
      style="white-space: nowrap; height: 100%; width: 100px;"
    >
      <div
        class="css-16inpur"
      >
        <div
          class="css-krjmb"
        >
          <button
            aria-controls="disclosure-:r0:"
            aria-expanded="true"
            class="chakra-button css-4xx2wk"
            data-testid="toggle-sidebar"
            type="button"
          >
            <div
              class="css-gmuwbf"
            >
              <svg
                fill="currentColor"
                height="24px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="chakra-stack css-1aad1bn"
        >
          <div
            class="css-8b158v"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M3 6a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v12a3 3 0 0 1 -3 3h-12a3 3 0 0 1 -3 -3v-12z"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l5 5"
                />
                <path
                  d="M14 14l1 -1c.928 -.893 2.072 -.893 3 0l3 3"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              My NFTs
            </p>
          </div>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1ej9zno"
          />
          <div
            class="css-21y338"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M11 21h-5a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v5.5"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l2 2"
                />
                <path
                  d="M17.8 20.817l-2.172 1.138a.392 .392 0 0 1 -.568 -.41l.415 -2.411l-1.757 -1.707a.389 .389 0 0 1 .217 -.665l2.428 -.352l1.086 -2.193a.392 .392 0 0 1 .702 0l1.086 2.193l2.428 .352a.39 .39 0 0 1 .217 .665l-1.757 1.707l.414 2.41a.39 .39 0 0 1 -.567 .411l-2.172 -1.138z"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              Favorites NFTs
            </p>
          </div>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1ej9zno"
          />
          <div
            class="css-21y338"
            role="group"
          >
            <div
              class="css-1jour5x"
            >
              <svg
                fill="none"
                height="24px"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M15 8h.01"
                />
                <path
                  d="M12 21h-6a3 3 0 0 1 -3 -3v-12a3 3 0 0 1 3 -3h12a3 3 0 0 1 3 3v6"
                />
                <path
                  d="M3 16l5 -5c.928 -.893 2.072 -.893 3 0l3 3"
                />
                <path
                  d="M14 14l1 -1c.48 -.461 1.016 -.684 1.551 -.67"
                />
                <path
                  d="M19.001 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
                />
                <path
                  d="M19.001 15.5v1.5"
                />
                <path
                  d="M19.001 21v1.5"
                />
                <path
                  d="M22.032 17.25l-1.299 .75"
                />
                <path
                  d="M17.27 20l-1.3 .75"
                />
                <path
                  d="M15.97 17.25l1.3 .75"
                />
                <path
                  d="M20.733 20l1.3 .75"
                />
              </svg>
            </div>
            <p
              class="chakra-text css-1qhmto6"
            >
              Observed Collections
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8atqhb"
    >
      <div
        data-testid="children"
      >
        children
      </div>
    </div>
  </div>
</div>
`;
